### ISSUE TITLE
A few minor questions regarding run_summarization.py in examples

### ISSUE THREAD

**Issue by JJumSSu** [huggingface/transformers#15347]
Hi, thank you for the amazing repo.

I'm currently using the script `run_summarization.py` from examples (I'm using ver 4.15.0).
When I run the code below, it seems that my trained model is not using beam search during generation.
Am I correct?

```
python examples/pytorch/summarization/run_summarization.py \
    --model_name_or_path t5-small \
    --do_train \
    --do_eval \
    --dataset_name cnn_dailymail \
    --dataset_config "3.0.0" \
    --source_prefix "summarize: " \
    --output_dir /tmp/tst-summarization \
    --per_device_train_batch_size=4 \
    --per_device_eval_batch_size=4 \
    --overwrite_output_dir \
    --predict_with_generate
```

Furthermore, when I set 'evaluation_strategy' to either `epochs `or `steps `, the performance of the validation set during training and the final performance of the validation set measured at the end of training (`do_eval`) seems quite different. 
Is this behavior expected? or am I missing something (such as only using beam search for the final measurement).

Thank you!

**Comment by JJumSSu** [comment]
Also, what does `task_specific_params `do during running the script `run_summarization.py`?
When I am initializing the T5-model, the model configuration includes such parameters.
```
 "task_specific_params": {
    "summarization": {
      "early_stopping": true,
      "length_penalty": 2.0,
      "max_length": 200,
      "min_length": 30,
      "no_repeat_ngram_size": 3,
      "num_beams": 4,
      "prefix": "summarize: "
    }
```
